src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -33,12 +33,25 @@ public BuildEventArgsReader(BinaryReader binaryReader)
         }
 
         /// <summary>
+        /// Raised when the log reader encounters a binary blob embedded in the stream.
+        /// The arguments include the blob kind and the byte buffer with the contents.
+        /// </summary>
+        internal event Action<BinaryLogRecordKind, byte[]> OnBlobRead;
+
+        /// <summary>
         /// Reads the next log record from the binary reader. If there are no more records, returns null.
         /// </summary>
         public BuildEventArgs Read()
         {
             BinaryLogRecordKind recordKind = (BinaryLogRecordKind)ReadInt32();
 
+            while (IsBlob(recordKind))
```

Comment:
Think of it as `if (IsBlob(recordKind))`. If the current record is a blob, do the work. Then read again, and if it's another blob, do the work again. Read through all the blobs until we hit a proper record again. So basically "skip over a potential series of blobs".